api/python/provisioner/commands/validator/validator.py

```diff
@@ -255,3 +272,240 @@ class YumRepoDataValidator(DirValidator):
     def __init__(self):
         super().__init__({Path("repomd.xml"): FileValidator(required=True)},
                          required=True)
+
+
+@attr.s(auto_attribs=True)
+class FileScheme(ABC):
+
+    """
+    Abstract file scheme class.
+
+    It can be used to define an abstract interface if necessary.
+    Also it is useful to have the parent class for all file schemes inherited
+    from this one for comparison using `isinstance`, `type` and for `attr`
+    module.
+
+    Attributes
+    ----------
+    _unexpected_attributes: dict
+        Stores all attribute values that were not listed
+        in the class definition.
+
+        Note: This attribute will be useful in the implementation of logic
+        if we need to validate that some attributes should not be listed
+        in the data scheme.
+    """
+
+    # To avoid exception
+    #  <ValueError: No mandatory attributes allowed after an attribute with a
+    #   default value or factory.>
+    #  this attribute is not optional and doesn't have a default value
+    _unexpected_attributes: dict = attr.ib(
```

Comment:
It is convenient because the class-inheritance help us here and `from_args` method is reusable for other child classes. 
I improved that using the third method.